### ISSUE TITLE
[css-overflow-3] Scrollable Overflow contributions of zero height/width elements

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#4791]
It appears that, at least in some implementations, zero-width elements do not contribute to the horizontal overflow, but zero-height elements do contribute to the vertical overflow.

So overall it seems that zero-sized things contribute in the block axis but not the inline axis.

This is kinda weird, but if it's required for Web-compat we should spec it.

**Comment by fantasai** [comment]
Testcase A (block-level): http://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0A.one%20%7B%20width%3A%20120vw%3B%20height%3A%200%3B%20%7D%0A.two%20%7B%20width%3A%200%3B%20height%3A%20120vh%3B%20%7D%0A%3C%2Fstyle%3E%0A%3Cp%20class%3D%22one%22%3E%0A%3Cp%20class%3D%22two%22%3E


Testcase B (abspos): http://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0A.one%20%7B%20position%3A%20absolute%3B%20width%3A%20120vw%3B%20%7D%0A.two%20%7B%20position%3A%20absolute%3B%20height%3A%20120vh%3B%20%7D%0A%3C%2Fstyle%3E%0A%3Cp%20class%3D%22one%22%3E%0A%3Cp%20class%3D%22two%22%3E

**Comment by bfgeek** [comment]
Specifically for the inflow case this is affecting the "inflow" bounds, the element isn't directly contributing overflow itself. You can test this by adding a transform to both elements - neither affect the scrollable area.

**Comment by fantasai** [comment]
OK, 1. so the testcases need to add html, body { height: 0; } to work properly and 2. we need to spec that boxes with empty area do not contribute to scrollable overflow. If I understood @bfgeek correctly?

**Comment by bfgeek** [comment]
Thats correct. At least "directly" to scrollable overflow. They can *indirectly* contribute to scrollable overflow via something like:

```html
<!DOCTYPE html>
<div style="width: 100px; height: 100px; overflow: scroll; display: grid;">
  <div style="width: 0; height: 120px; outline: solid red;"></div>
  <div style="width: 120px; height: 0; outline: solid green;"></div>
</div>
```

**Comment by fantasai** [comment]
@bfgeek Edited into https://drafts.csswg.org/css-overflow-3/#scrollable
Let us know if that looks to be correct?

**Comment by MrHBS** [comment]
@fantasai Is there anything left with this issue?

**Comment by fantasai** [comment]
@MrHBS No, just wanted to ask @bfgeek to sign off that the edits are correct.
